File: Cpp/BaekJoon/Sliver/S2_6603.cpp
#include <iostream>
#include <vector>

using namespace std;

void backtracking(int k, vector<int>& nums, vector<int>& path, int now,
                  int idx) {
  if (now == 6) {
    for (int i = 0; i < 6; i++) {
      cout << path[i] << " ";
    }
    cout << "\n";

    return;
  }

  for (int i = idx; i < nums.size(); i++) {
    path.push_back(nums[i]);

    backtracking(k, nums, path, now + 1, i + 1);

    path.pop_back();
  }

  return;
}

int main() {
  while (true) {
    int n;
    cin >> n;

    if (n == 0) break;

    vector<int> nums(n);
    for (int i = 0; i < n; i++) {
      cin >> nums[i];
    }

    vector<int> path;
    backtracking(n, nums, path, 0, 0);

    cout << "\n";
  }

  return 0;
}

File: Java/BaekJoon/Silver/S2_6603.java
package BaekJoon.Silver;

import java.util.Scanner;

public class S2_6603 {
    public static void backtracking(int k, int[] nums, int[] path, int now, int idx) {
        if (now == 6) {
            for (int i = 0; i < 6; i++) {
                System.out.print(path[i] + " ");
            }

            System.out.println();

            return;
        }

        for (int i = idx; i < nums.length; i++) {
            path[now] = nums[i];

            backtracking(k, nums, path, now + 1, i + 1);

            path[now] = -1;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            int n = sc.nextInt();

            if (n == 0)
                break;

            int[] nums = new int[n];
            for (int i = 0; i < n; i++) {
                nums[i] = sc.nextInt();
            }

            int[] path = new int[6];

            backtracking(n, nums, path, 0, 0);

            System.out.println();
        }

        sc.close();

        return;
    }
}
